e2e/test/scenarios/dashboard/tabs.cy.spec.js

```diff
@@ -486,10 +530,137 @@ describe("scenarios > dashboard > tabs", () => {
+    });
+  });
+
+  it("should only fetch cards on the current tab of an embedded dashboard", () => {
+    cy.intercept("PUT", "/api/dashboard/*").as("saveDashboardCards");
+    cy.intercept("PUT", "/api/dashboard/*").as("saveDashboardCards");
+    cy.intercept("POST", "/api/card/*/query").as("cardQuery");
+
+    visitDashboardAndCreateTab({
+      dashboardId: ORDERS_DASHBOARD_ID,
+      save: false,
+    });
+
+    // Add card to second tab
+    cy.icon("pencil").click();
+    openQuestionsSidebar();
+    sidebar().within(() => {
+      cy.findByText("Orders, Count").click();
+    });
+
+    cy.wait("@cardQuery");
+
+    saveDashboard();
+
+    cy.wait("@saveDashboardCards").then(({ response }) => {
+      cy.wrap(response.body.dashcards[1].id).as("secondTabDashcardId");
+    });
+
+    const firstQuestion = () => {
+      return cy.request("GET", `/api/card/${ORDERS_QUESTION_ID}`).its("body");
+    };
+    const secondQuestion = () => {
+      return cy
+        .request("GET", `/api/card/${ORDERS_COUNT_QUESTION_ID}`)
+        .its("body");
+    };
+
+    firstQuestion().then(r => {
+      expect(r.view_count).to.equal(1); // 1 (firstQuestion)
+    });
+    secondQuestion().then(r => {
+      expect(r.view_count).to.equal(1); // 1 (secondQuestion)
+    });
+
+    cy.intercept(
+      "GET",
+      `/api/embed/dashboard/*/dashcard/*/card/${ORDERS_QUESTION_ID}`,
+      cy.spy().as("firstTabQuery"),
+    );
+    cy.intercept(
+      "GET",
+      `/api/embed/dashboard/*/dashcard/*/card/${ORDERS_COUNT_QUESTION_ID}`,
+      cy.spy().as("secondTabQuery"),
+    );
+
+    openStaticEmbeddingModal({ activeTab: "parameters", acceptTerms: true });
+
+    // publish the embedded dashboard so that we can directly navigate to its url
+    publishChanges("dashboard", ({ request }) => {});
+    // directly navigate to the embedded dashboard, starting on Tab 1
+    visitIframe();
+    // wait for results
+    cy.findAllByTestId("dashcard").contains("37.65");
+    cy.signInAsAdmin();
+    cy.get("@firstTabQuery").should("have.been.calledOnce");
+    cy.get("@secondTabQuery").should("not.have.been.called");
+
+    cy.wait(500); // this is needed to avoid flakes
+    firstQuestion().then(r => {
+      expect(r.view_count).to.equal(3); // 1 (previously) + 1 (firstQuestion) + 1 (first tab query)
+    });
+    secondQuestion().then(r => {
+      expect(r.view_count).to.equal(2); // 1 (previously) + 1 (secondQuestion)
+    });
+
+    goToTab("Tab 2");
+    cy.get("@secondTabQuery").should("have.been.calledOnce");
+    cy.get("@firstTabQuery").should("have.been.calledOnce");
+
+    cy.wait(500); // this is needed to avoid flakes
+    firstQuestion().then(r => {
+      expect(r.view_count).to.equal(4); // 3 (previously) + 1 (firstQuestion)
+    });
+    secondQuestion().then(r => {
+      expect(r.view_count).to.equal(4); // 2 (previously) + 1 (secondQuestion) + 1 (second tab query)
+    });
+
+    goToTab("Tab 1");
+    cy.wait(1000);
```

Comment:
Wdyt about replacing cy`.wait` with finding elements on the page that will indicate the state is ready for the assertion? Or waiting for requests being resolved